kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/collections/CollectionMatchers.kt

```diff
@@ -176,41 +175,77 @@ fun <T> matchEach(expected: List<T>, asserter: (T, T) -> Unit): Matcher<Collecti
       }
    })
 
+private sealed interface MatchEachProblem {
+   val atIndex: Int
+   val shortMessage: String?
+   val longMessage: String?
+
```

Comment:
Can we make these not null, and construct some relevant message when we have `Thrown` and there's no message/stacktrace?

Reply:
Good point, fixed.